packages/api/amplify_api_dart/lib/src/graphql/factories/graphql_request_factory.dart

```diff
@@ -98,8 +108,35 @@ class GraphQLRequestFactory {
     // build inputs based on request operation
     switch (operation) {
       case GraphQLRequestOperation.get:
-        upperOutput = r'($id: ID!)';
-        lowerOutput = r'(id: $id)';
+        final indexes = schema.indexes;
+        final modelIndex =
+            indexes?.firstWhereOrNull((index) => index.name == null);
+        if (modelIndex != null) {
+          // custom index(es), e.g.
+          // upperOutput: ($customId: ID!, name: String!)
+          // lowerOutput: (customId: $customId, name: $name)
+          final bUpperOutput = StringBuffer(_openParen);
+          final bLowerOutput = StringBuffer(_openParen);
+          for (final field in modelIndex.fields) {
+            final isId = field == modelIndex.fields.first;
+            final graphQLTypeString =
+                isId ? 'ID' : getGraphQLTypeForField(schema, field);
```

Comment:
Per offline discussion, added comment to address this. Bc codegen doesn't capture scalar type in schema, will populate the document string with values instead of referencing variables (which requires scalar type declared) in the case of custom index in get() query.